### PR TITLE
add OpenAIChat->generateChat and Message helper

### PR DESCRIPTION
**Pull request by chr-hertel**
Missed that, what do you think?

### CODE REVIEW

File: src/Chat/Message.php [hunk 1]
```diff
@@ -9,4 +9,40 @@ final class Message
...
+    public static function assistant(string $content): self
+    {
+        $message = new self();
+        $message->role = ChatRole::Assistant;
+        $message->content = $content;
+
+        return $message;
+    }
+
+    public static function function(string $content): self
```
Comment (MaximeThoonsen) [inline src/Chat/Message.php:40]: what would be the usage of the function "function" ? and maybe not use the word function in case it breaks some tools.

**Comment by MaximeThoonsen** [comment]
I created a Chat class and forget to create the chat function. 😂
Thanks @chr-hertel. 

I think we don't need the function function because we have this one:     

`
public function addFunction(FunctionInfo $functionInfo): void
    {
        $this->functions[] = $functionInfo;
    }
   `
   
   WDYT ?

**Comment by chr-hertel** [comment]
well, i was unsure about the necessity of that `Message::function(...)` as well, but thought it is mostly for a function's result, right? so not for registering a function, but adding it's output to the message collection.

**Comment by MaximeThoonsen** [comment]
Very good point, all my use case the function was kinda the final steps but we should be able to continue the conversation.
So let's add it, but I'm not super fan of the function called "function" 😅
You can name it "functionCall" like in [the API](https://platform.openai.com/docs/api-reference/chat/create)

**Comment by chr-hertel** [comment]
Yea, get it, what about `Message::functionResult(string $content)` since it's more the result of a function call, is it?

but no real preference - your call :)

**Comment by MaximeThoonsen** [comment]
We won't pass the result of the function but the description of the call. What about FunctionCalled ?

**Comment by chr-hertel** [comment]
oh, i'm afraid we have a different understanding about function calling than

this is an example from the introducing blog post:
```json
  "messages": [
    {"role": "user", "content": "What is the weather like in Boston?"},
    {"role": "assistant", "content": null, "function_call": {"name": "get_current_weather", "arguments": "{ \"location\": \"Boston, MA\"}"}},
    {"role": "function", "name": "get_current_weather", "content": "{\"temperature\": "22", \"unit\": \"celsius\", \"description\": \"Sunny\"}"}
  ],
```
see https://openai.com/blog/function-calling-and-other-api-updates

so my understanding is, that after a function call the output of the function execution is added as separate message to the collection of messages and provided back to GPT.

this is an example how i implemented it a few weeks back:
https://github.com/chr-hertel/symfonylive-berlin-bot/blob/function-calling/src/ChatBot/LlmChain/FunctionChain.php

File: src/Chat/Message.php (hunk 1 above)
Comment (chr-hertel) [reply src/Chat/Message.php]: renamed it to `functionCall` :)

**Comment by MaximeThoonsen** [comment]
Ohhh you are right .. again!! Sorry I never dug that part. So yeah functionResult is way better. I'll rename it so I don't bother you another time

**Comment by chr-hertel** [comment]
No, don't worry at all - I guess we're learning here together. Thanks for taking care - happy to contribute!

**Comment by MaximeThoonsen** [comment]
I added you to the team with the "Write" rights. Feel free to work on this repo as it was yours

**Comment by chr-hertel** [comment]
Thanks man, happy to do so! :)

**Pull request merged**